hypex/transformers/abstract.py

```diff
@@ -1,22 +1,29 @@
 from abc import abstractmethod
 from typing import Any
 
-from hypex.dataset import Dataset
+from hypex.dataset import Dataset, ExperimentData, ABCRole
 from hypex.executor import Calculator
 from hypex.utils import AbstractMethodError
 
 
 class Transformer(Calculator):
     @property
-    def __is_transformer(self):
+    def _is_transformer(self):
         return True
 
-    @classmethod
+    @staticmethod
     @abstractmethod
+    def _inner_function(data: Dataset, **kwargs) -> Dataset:
+        raise AbstractMethodError
+
+    @classmethod
     def calc(cls, data: Dataset, **kwargs):
         return cls._inner_function(data, **kwargs)
 
+    def execute(self, data: ExperimentData) -> ExperimentData:
+        data = data.copy(data=self.calc(data=data.ds))
+        return data
+
     @staticmethod
-    @abstractmethod
-    def _inner_function(data: Dataset, **kwargs) -> Any:
-        raise AbstractMethodError
+    def _list_unification(roles: Any):
```

Comment:
leave this method into Adapter like: Adapter.to_list(value)

Reply:
fixed